### PR TITLE
Petter od authlog

### PR DESCRIPTION
**Pull request by pettersv**
**What this PR does / why we need it**:
The authlog (/var/log/authlog) contains information about failed access attempts to the nodes. This PR tails the authlog, formats the message and sends it to opendistro.

**NOTE that in order for this to work you need to use version 1.12.1 or later of the elastisys/opendistro-es chart** 

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #140

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

**Comment by viktor-f** [comment]
As far as I can see this looks good. I guess you also need to add a policy for the new index here: https://github.com/elastisys/opendistro-build/tree/main/helm/opendistro-es/files/configurer/ism-policies

File: helmfile/values/fluentd-user.yaml.gotmpl [hunk 1]
```diff
@@ -62,6 +62,19 @@ extraConfigMaps:
         </pattern>
       </parse>
     </source>
+    # Index the authlog
```
Comment (OlleLarsson) [inline helmfile/values/fluentd-user.yaml.gotmpl:65]:
```suggestion
    # Tail authlog
```

File: helmfile/values/fluentd-user.yaml.gotmpl [hunk 2]
```diff
@@ -62,6 +62,19 @@ extraConfigMaps:
...
+    <source>
+      @id authlog
+      @type tail
+      path /var/log/auth.log
+      pos_file /var/log/auth.pos
+      tag authlog.*
```
Comment (OlleLarsson) [inline helmfile/values/fluentd-user.yaml.gotmpl:71]: We could do without the `.*` if we want. All it does is add the add the dotnotation path, e.g. events will be emitted with tag `authlog.var.log.auth.log`.

File: helmfile/values/fluentd-user.yaml.gotmpl [hunk 3]
```diff
@@ -120,6 +133,44 @@ extraConfigMaps:
     @include /etc/fluent/extra-config.d/*.conf
 
   output.conf: |-
+    <match authlog.**>
+       @id elasticsearch_authlog
+       @type elasticsearch
+       @log_level "#{ENV['OUTPUT_LOG_LEVEL']}"
+       include_tag_key true
```
Comment (OlleLarsson) [inline helmfile/values/fluentd-user.yaml.gotmpl:140]: Seems kind of unessecary to include the tag as a field.

**Review by OlleLarsson — COMMENTED** [review]
Good job.
Does the logs inlcude any timestaps?
I'm wondering if we should use that as [time_key](https://github.com/uken/fluent-plugin-elasticsearch#time_key) or not :thinking:

File: helmfile/values/fluentd-user.yaml.gotmpl (hunk 1 above)
Comment (pettersv) [reply helmfile/values/fluentd-user.yaml.gotmpl:65]: Fixed :)

File: helmfile/values/fluentd-user.yaml.gotmpl (hunk 2 above)
Comment (pettersv) [reply helmfile/values/fluentd-user.yaml.gotmpl:71]: We seem to use dotnotation for the other indicies so I'm leaving it for now.

File: helmfile/values/fluentd-user.yaml.gotmpl (hunk 3 above)
Comment (pettersv) [reply helmfile/values/fluentd-user.yaml.gotmpl:140]: Yes, removed it

**Comment by OlleLarsson** [comment]
Remember to add configuration options for the retention of authlog indices.

File: helmfile/values/fluentd-wc.yaml.gotmpl [hunk 4]
```diff
@@ -65,6 +65,18 @@ extraConfigMaps:
       @type record_transformer
       remove_keys responseObject,requestObject
     </filter>
+    # Index the authlog
```
Comment (OlleLarsson) [inline helmfile/values/fluentd-wc.yaml.gotmpl:68]:
```suggestion
    # Tail the authlog
```

**Review by OlleLarsson — COMMENTED** [review]
Still missing options to configure [retention](https://github.com/elastisys/opendistro-build/blob/main/helm/opendistro-es/values.yaml#L614-L615) for authlog indices.

**Comment by pettersv** [comment]
> Still missing options to configure [retention](https://github.com/elastisys/opendistro-build/blob/main/helm/opendistro-es/values.yaml#L614-L615) for authlog indices.

OK so wherer does this belong? In sc-config?

**Review by OlleLarsson — APPROVED** [review]
LGTM.
Have you tried upgrading to see that everything works as expected, that configurer creates the new resources for authlogs etc? I personally think that it will create everything needed but just checking if you've tested.

**Comment by pettersv** [comment]
> LGTM.
> Have you tried upgrading to see that everything works as expected, that configurer creates the new resources for authlogs etc? I personally think that it will create everything needed but just checking if you've tested.

Yes, I have tested it. Although I did not set up a pristine cluster for all the tests, did some manual cleaning to save time.

**Review by viktor-f — APPROVED** [review]
Looks good.
Though the retention values seem a bit strange. In particular that the default size is higher that the prod size. So change that if you want, but it's no blocker.

**Comment by pettersv** [comment]
> Looks good.
> Though the retention values seem a bit strange. In particular that the default size is higher that the prod size. So change that if you want, but it's no blocker.

I copied the values from the "other*" index as I have no real clue what it should be. We have to adjust this later once we get the feeling for how large this index grows. Btw the other* index has the same strange setup with larger values in default than prod. I'll create an issue for that.

**Pull request merged**